portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,9 +92,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = request.getParameter("comment-input");
-    String name = request.getParameter("name-input");
     long timestamp = System.currentTimeMillis();
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+    String id = userService.getCurrentUser().getUserId();
+    String name = LoginServlet.getUserNickname(id);
```

Comment:
If the user hasn't set a nickname, their name will be set to the empty string. I'm wondering if there is a more useful piece of information you could use instead -- for example, their ID? Or would that not exist also?

Reply:
Thanks for your suggestions! The comment I had to describe this function was inaccurate, it only returns an empty string if the user's id isn't in datastore.